src/main/java/org/testng/remote/RemoteTestNG.java

```diff
@@ -168,14 +168,7 @@ public static void main(String[] args) throws ParameterException {
     }
     m_debug = cla.debug;
     m_ack = ra.ack;
-    if (m_debug) {
-//      while (true) {
-        initAndRun(args, cla, ra);
-//      }
-    }
-    else {
-      initAndRun(args, cla, ra);
-    }
+    initAndRun(args, cla, ra);

```

Comment:
This one should stay as it is since it makes it easy to switch `m_debug` on or off. Your change completely removes this.